### DISCUSSION TITLE
Why does DIA-NN library-free run so much slower vs. DIA-NN with an input spectral library?

### DISCUSSION THREAD

**Discussion by ec-ho-ra-mos** [vdemichev/DiaNN#811] [General]
I noticed that DIA-NN 1.8.1, when run library-free (i.e., using an input uniprot.fasta file), takes about 5 hours on a single thread. However, when I use a spectral library, it only takes about 3 minutes. I understand that library-free DIA-NN involves spectral library prediction (i.e., Prosit RT and MS2 prediction), which adds extra time. But is that the only difference? I would love to know more about what other steps are skipped when DIA-NN is run with an input library.

**Comment by vdemichev** [comment]
Yes, this can be the case, lib-free means huge search space. I can also take a look at the logs if you wish.
